### PR TITLE
CV: skip sporadically failing template delivery

### PR DESCRIPTION
**Pull request by mbien**
https://github.com/apache/netbeans/pull/7333 degrades CV test stability

 - failure: ["FSException: Cannot get shared access"](https://github.com/apache/netbeans/pull/7333#issuecomment-2082941539)
 - tests can be brute forced into passing.
 - It is not clear if something is wrong with the hint editor support or if this template is just degrading the sporadically failing tests further due to changed timing.

lets try to skip over it and check if it improves the stability.

other:

 - wrap `masterfs.macosx` in retry script due to sporadic failures in `OSXNotifierTest#testNextEvent`
 - `NbTestCase`: use `Files#delete` for better exception on failure, (second commit)
   - "Cannot delete file" isn't very useful, `Files#delete` will state the cause if available
   - might help to debug some sporadic test failures
   - global refactoring to replace similar patterns resulted in some `TestUtil` cleanup which was duplicated everywhere

### CODE REVIEW

**Comment by mbien** [comment]
so the CV tests on linux passed on first attempt, which is a good sign (esp when compared to https://github.com/apache/netbeans/actions/runs/8876164558).

The macos networking test seems to fail often recently, but I can't help with that one due to lack of hardware + it isn't related to this problem. I might wrap it in the retry script.

File: platform/openide.util.ui/test/unit/src/org/openide/util/test/TestFileUtils.java [hunk 1]
```diff
@@ -48,52 +46,48 @@ public class TestFileUtils {
...
-        while ((read = is.read(buf)) != -1) {
-            baos.write(buf, 0, read);
+        return Files.readAllBytes(file.toPath());
+    }
+
+    /**
+     * Deletes the file or directory and its contents.
+     */
+    public static void deleteFile(File file) throws IOException {
+        if (file.isDirectory() && file.equals(file.getCanonicalFile())) {
```
Comment (matthiasblaesing) [inline platform/openide.util.ui/test/unit/src/org/openide/util/test/TestFileUtils.java:85]: What is the reason for the second condition?

**Review by matthiasblaesing — APPROVED** [review]
Of course a fix for the root cause of the instability would be preferred, but not causing false-positives is more important.
In general this looks like a good cleanup. Thank you!

File: platform/openide.util.ui/test/unit/src/org/openide/util/test/TestFileUtils.java (hunk 1 above)
Comment (mbien) [reply platform/openide.util.ui/test/unit/src/org/openide/util/test/TestFileUtils.java:85]: I believe this is a way of checking that the recursion didn't walk through a symlink and is clearing half of the drive.

Please note that I copied this method mostly from the other util class (there were 15 variants of this method copied around). I tried to centralize it and made the actual delete operation use `Files#delete` to get a cause msg next time it fails in CI.

I thought about reimplementing this fully using `Path`, but keeping it as-is felt like the less risky approach for a hotfix. But now that I know that there is a shared test utility class, this opens up more cleanup and deduplicaiton in future.

Comment (mbien) [reply platform/openide.util.ui/test/unit/src/org/openide/util/test/TestFileUtils.java:85]: https://github.com/search?q=repo%3Aapache%2Fnetbeans%20%22file.isDirectory()%20%26%26%20file.equals(file.getCanonicalFile())%22&type=code more copies of this exact variant

Comment (matthiasblaesing) [reply platform/openide.util.ui/test/unit/src/org/openide/util/test/TestFileUtils.java:85]: A thanks - kind of makes sense. Indeed I though, that `File.walk` might make more sense here.

Comment (mbien) [reply platform/openide.util.ui/test/unit/src/org/openide/util/test/TestFileUtils.java:85]: I will put this on my cleanup todo list

Comment (mbien) [reply platform/openide.util.ui/test/unit/src/org/openide/util/test/TestFileUtils.java:85]: or let me implement this properly right away, why not

**Review by matthiasblaesing — APPROVED** [review]
Thanks!

**Pull request merged**